workspaces-tests/src/bin/payroll_example.rs

```diff
@@ -0,0 +1,109 @@
+use near_contract_tools::{
+    rbac::{self, Rbac},
+    Rbac,
+};
+use near_sdk::{
+    borsh::{self, BorshDeserialize, BorshSerialize},
+    collections::UnorderedMap,
+    env, ext_contract, near_bindgen, require, AccountId, BorshStorageKey, IntoStorageKey,
+    PanicOnDefault, Promise,
+};
+
+enum PayrollKey {
+    LOG,
+    FEE,
+}
+
+impl IntoStorageKey for PayrollKey {
+    fn into_storage_key(self) -> Vec<u8> {
+        match self {
+            PayrollKey::LOG => b"~pl".to_vec(),
+            PayrollKey::FEE => b"~pf".to_vec(),
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize, BorshStorageKey)]
+enum Role {
+    Manager,
+    Employee,
+}
+
+#[derive(BorshSerialize, BorshDeserialize, PanicOnDefault, Rbac)]
+#[rbac(roles = "Role")]
+#[near_bindgen]
+struct Payroll {
+    pub hourly_fee: u32,
+    pub logged_time: UnorderedMap<AccountId, u8>,
+}
+
+/// Manager can add new employees and disburse payments
+/// Employees can log time
+#[near_bindgen]
+impl Payroll {
+    pub fn new() -> Self {
+        Payroll {
+            hourly_fee: 1000,
+            logged_time: UnorderedMap::new(PayrollKey::LOG),
+        }
+    }
+
+    /// Manager can add new employees
+    pub fn add_employee(&mut self, account_id: &AccountId) {
+        self.require_role(&Role::Manager);
+        self.logged_time.insert(account_id, &0);
+
+        // write updated time log to state
+        env::state_write(&self.logged_time);
```

Comment:
This overwrites the default struct.

Reply:
I've fixed this now by writing the whole contract state `env::state_write(self)`. However this is feels slightly off because say for a contract with many collections I only want to update one of the collections then this won't work.

Even [near collections docs](https://docs.near.org/sdk/rust/contract-structure/collections#error-prone-patterns) don't talk about this and [storage_write](https://docs.rs/near-sdk/latest/near_sdk/env/fn.storage_write.html) which I believe should allow writing selective collections is non-intuitive to use.

Do you this example should show how to write a specific collection to state?